rustls/src/client/hs.rs

```diff
@@ -146,16 +147,18 @@ pub(super) fn start_handshake(
     let random = Random::new(config.provider.secure_random)?;
     let extension_order_seed = crate::rand::random_u16(config.provider.secure_random)?;
 
-    // TODO(XXX): Construct ECH context state, use to perform ECH flow when
-    //   emitting client hello.
-    let _ech_context = config
-        .ech_config
-        .as_ref()
-        .map(|ech_config| {
-            let _suite = &ech_config.suite;
-            let _config = &ech_config.config;
-            Some(())
-        });
+    let ech_state = match config.ech_config.as_ref() {
```

Comment:
Hmm, I think it might make things more awkward? I can't use an `if let / else` to match the `&EchMode` given to the `map` closure because of MSRV, so I'd still need a `match { }` in the `map` body making it about the ~same amount of code, but more annoying to deal with the fallibility of the `EchState::new()`.

Am I overthinking what you had in mind?

Reply:
I missed the `?` operator. I can see how that makes this probably not worth it.